lib/stitches/auto_run.py

```diff
@@ -0,0 +1,229 @@
+
+    path = find_path(graph, starting_point, ending_point)
+    path = add_path_attribs(path)
+
+    new_elements, original_parents = path_to_elements(graph, path)
+
+    if preserve_order:
+        for element, parent in zip(new_elements, original_parents):
+            if parent is not None:
+                parent.append(element)
+                element.set('transform', get_correction_transform(parent, child=True))
+    else:
+        parent = elements[0].node.getparent()
+        insert_index = parent.index(elements[0].node)
+        group = create_new_group(parent, insert_index, _("Auto-Run"))
+        insert_elements(group, new_elements)
+
+    remove_original_elements(elements)
+
+
+def build_graph(elements, preserve_order, break_up):
+    if preserve_order:
+        graph = nx.DiGraph()
+    else:
+        graph = nx.Graph()
+
+    for element in elements:
+        if not isinstance(element, Stroke):
+            continue
+
+        if not break_up:
+            segments = [LineString(path) for path in element.paths]
+        else:
+            segments = break_up_segments(element, elements)
+
+        for segment in segments:
+            for c1, c2 in zip(segment.coords[:-1], segment.coords[1:]):
+                start = Point(*c1)
+                end = Point(*c2)
+
+                graph.add_node(str(start), point=start)
+                graph.add_node(str(end), point=end)
+                graph.add_edge(str(start), str(end), element=element)
+
+                if preserve_order:
+                    # The graph is a directed graph, but we want to allow travel in
+                    # any direction, so we add the edge in the opposite direction too.
+                    graph.add_edge(str(end), str(start), element=element)
+
+    return graph
+
+
+def break_up_segments(element, elements):
+    '''
+    Add extra nodes at intersections and nearest points.  Users will still have control
+    over the precision by adding nodes where they think good breaking points are.
+    '''
+    segment_list = []
+    line_strings = [LineString(path) for path in element.paths]
+    for line in line_strings:
+        points = []
+        # add points at intersections with other elements (not at self intersections, sorry)
+        intersection_points = get_intersections(line, element, elements)
+        if intersection_points:
+            points.extend(intersection_points)
+        # split at points
+        points = MultiPoint(points)
+        # split line at points, trouble: split() doesn't seem to work on small lines at all
+        # but differene with buffer seems to do the trick
+        # source: https://gis.stackexchange.com/questions/297134/shapely-floating-problems-with-split/327287#327287
+        seg = line.difference(points.buffer(1e-13))
+        if isinstance(seg, LineString):
+            segment_list.append(seg)
+        else:
+            for geom in seg.geoms:
+                if isinstance(geom, LineString) and geom.length > 1:
```

Comment:
Why `> 1`?  That's 1px, which is about 0.26mm.  Seems like we might care about smaller pieces, since most embroidery machines have 0.1mm resolution?